src/base/bittorrent/peerinfo.cpp

```diff
@@ -198,7 +198,16 @@ QString PeerInfo::I2PAddress() const
 
 QString PeerInfo::client() const
 {
-    return QString::fromStdString(m_nativeInfo.client);
+    QString client = QString::fromStdString(m_nativeInfo.client);
```

Comment:
@HanabishiRecca 
I really suggest you write a loop (over `m_nativeInfo.client`) and do all the filtering and mapping there. It is clear that `m_nativeInfo.client` is `std::string` which is `std::basic_string<char>` which means you won't be dealing with utf-8 characters. You could even use `std::isprint` for detection before converting to `QChar` to save some cpu cycles.

Reply:
Well forget this.

Reply:
> I really suggest you write a loop (over `m_nativeInfo.client`) and do all the filtering and mapping there.

Are you sure? Aren't there multibyte non-printable characters in UTF-8? Or do you suggest to handle them manually?

Reply:
Yeah, it's impossible to mutate UTF-8 characters in-place, as they are variable length.

In contrast, `QChar` is a UTF-16 character with fixed size, which is advantageous for this scenario.

Reply:
I already ditched/withdraw this discussion https://github.com/qbittorrent/qBittorrent/pull/20788#discussion_r1587408231